### PR TITLE
Add resource requests to addons

### PR DESCRIPTION
**Pull request by gmarek**
This is first part of fix for #18211

I don't know how we missed this, but I believe it's a bug.

cc @wojtek-t @fgrzadkowski @bgrant0607 @brendandburns @thockin @davidopp

### CODE REVIEW

**Comment by vishh** [comment]
Why is this PR needed? The defaulting logic will default `requests` to `limits`.

**Comment by gmarek** [comment]
Wasn't it supposed to mean 'burstable' QoS? I.e. we guarantee nothing, but allow usage of limit? Did it changed?

**Comment by gmarek** [comment]
cc @bgrant0607

**Comment by vishh** [comment]
If we removed `limit` and added `request`, then these pods will be in
`Burstable` class. The other option is to set `limit` to be greater than
`request`. IIUC, this PR does neither. Also, is it safe for some of the
addons to be `Burstable`? These addons will be more prone to evictions on
the node.

**Comment by gmarek** [comment]
OK, let me rephrase what I wanted to tell: shouldn't add-ons be in the highest (guaranteed IIRC the name) class, which was characterized by having `request=limit`?. @vishh - IIUC you depend on a backward-compatible hack assuming that if there's no request it's an 'old' definition from the time where there was no requests, right? I don't think we should stay that way.

**Comment by vishh** [comment]
Got it. My concern is that if someone else updates just the limits in the future, that will place these addons in the Burstable class. We can work around that with some comments possibly.
If the intention is to keep these addons int eh Guaranteed class, this PR LGTM.

**Comment by vishh** [comment]
Ping @gmarek

**Comment by gmarek** [comment]
I'm just waiting for LGTM or some other comment.

**Comment by vishh** [comment]
Can you add comments to ensure that the limits match requests over time?

**Comment by gmarek** [comment]
Sure - done. @vishh PTAL

**Comment by vishh** [comment]
Thanks @gmarek

**Comment by gmarek** [comment]
@k8s-bot unit test this
